packages/synapse-react-client/src/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,52 +1,30 @@
 import React, { useState } from 'react'
-import { Alert, Button, Link, TextField, Typography } from '@mui/material'
-import {
-  ChangePasswordWithCurrentPassword,
-  ChangePasswordWithTwoFactorAuthToken,
-  TwoFactorAuthErrorResponse,
-  TwoFactorAuthOtpType,
-} from '@sage-bionetworks/synapse-types'
+import { Alert, Button, Link, TextField } from '@mui/material'
 import { Link as RouterLink } from 'react-router-dom'
-import { ONE_TIME_PASSWORD_STEP, OneTimePasswordForm } from '../Authentication'
-import {
-  useChangePassword,
-  useGetCurrentUserProfile,
-} from '../../synapse-queries'
+import { useGetCurrentUserProfile } from '../../synapse-queries'
 import { displayToast } from '../ToastMessage'
-import {
-  RECOVERY_CODE_GUIDANCE_TEXT_SHORT,
-  TOTP_GUIDANCE_TEXT,
-} from '../Authentication/Constants'
+import useChangePasswordFormState from './useChangePasswordFormState'
 
 export default function ChangePassword() {
   const [oldPassword, setOldPassword] = useState<string>('')
   const [newPassword, setNewPassword] = useState<string>('')
   const [confirmPassword, setConfirmPassword] = useState<string>('')
-  const [otpStep, setOtpStep] =
-    useState<ONE_TIME_PASSWORD_STEP>('VERIFICATION_CODE')
-  const [twoFactorAuthErrorResponse, setTwoFactorAuthErrorResponse] = useState<
-    TwoFactorAuthErrorResponse | undefined
-  >()
 
   const { data: userProfile, isLoading: isLoadingUserProfile } =
     useGetCurrentUserProfile()
+
   const {
-    mutate: changePassword,
+    promptForTwoFactorAuth,
+    twoFactorAuthPrompt,
     isPending: changePasswordIsPending,
+    handleChangePasswordWithCurrentPassword,
     error,
-  } = useChangePassword({
-    onSuccess: maybeTwoFactorResponse => {
-      if (maybeTwoFactorResponse) {
-        // We store the 2FA Error Response in state because the `changePasswordResult` may change before we are done with the 2FA code,
-        // e.g. if the user enters the wrong 2FA code
-        setTwoFactorAuthErrorResponse(maybeTwoFactorResponse)
-      } else {
-        setTwoFactorAuthErrorResponse(undefined)
-        setOldPassword('')
-        setNewPassword('')
-        setConfirmPassword('')
-        displayToast('Password successfully changed.', 'success')
-      }
+  } = useChangePasswordFormState({
```

Comment:
Refactored to use this custom hook shared with new component in `ChangePasswordWithToken.tsx`